src/components/molecules/LoginForm/LoginForm.tsx

```diff
@@ -0,0 +1,139 @@
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import { useState, FC, FormEvent, ChangeEvent } from 'react';
+
+import Button, { EButtonTheme } from '@components/atoms/Button';
+import Input from '@components/atoms/Input';
+import { ETypographyVariant } from '@enums/typography';
+import { useAuth } from '@src/contexts/AuthContext';
+import { Auth } from '@types';
+
+import * as S from './LoginForm.styled';
+
+interface LoginProps {
+  type: string;
+}
+
+const LoginForm: FC<LoginProps> = ({ type }) => {
+  const { login, signup } = useAuth();
+  const router = useRouter();
+  const [loginData, setLoginData] = useState<Auth.ILoginData>({
+    email: '',
+    password: '',
+  });
+  const [signUpData, setSignUpData] = useState<Auth.ISignUpData>({
+    email: '',
+    password: '',
+    repeatPassword: '',
+  });
```

Comment:
Look at the changes to the `Auth` namespace. This change allows you to use one state object for both signup and login forms. And it also means that you can simplify the component below to conditionally render the repeatPassword input when the `authType` is `Signup`.
```suggestion
  const [formData, setFormData] = useState<Auth.IFormData>({
    email: '',
    password: '',
    repeatPassword: '',
  });
```

Hope that makes sense but we can catch up if you need some help